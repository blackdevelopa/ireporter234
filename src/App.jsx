import React from 'react';
import { BrowserRouter } from 'react-router-dom';

/**
 * @description - App component
 * @param {object} props
 * @returns {JSX} - Returns all Components
 */
const App = () => (
  <BrowserRouter>
    <div>
      <h1>Welcome to iReporter</h1>
    </div>
  </BrowserRouter>
);

export default App;
