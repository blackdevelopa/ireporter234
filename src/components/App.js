import React, { Component } from 'react';
import { BrowserRouter,  Route, Switch } from 'react-router-dom';
import LandingPage from './pages/Landing/LandingPage';
import './style/index.css';

class App extends Component {
  render() {
    return (
      <BrowserRouter>
        <Switch>
          <LandingPage />
        </Switch>
      </BrowserRouter>
    )
  }
}
export default App;
