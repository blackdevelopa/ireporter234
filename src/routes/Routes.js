import React, {Component} from 'react';
import { Route, Switch } from 'react-router-dom';
import LandingPage from '../components/pages/Landing/LandingPage';
import RedFlag from '../components/pages/RedFlag/RedFlag';
import Profile from '../components/pages/Profile/Profile';
import Intervention from '../components/pages/Intervention/Intervention';



class Routes extends Component {
  render () {
    return(
      <Switch>
        <Route path='/' component={LandingPage} exact />
        <Route path='/red-flags' component={RedFlag} />
        <Route path='/interventions' component={Intervention} />
        <Route path='/profile' component={Profile} />
      </Switch>
    )
  }
}


export default Routes;