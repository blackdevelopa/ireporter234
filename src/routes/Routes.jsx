import React from 'react';
import { Route, Switch } from 'react-router-dom';
import LandingPage from '../components/containers/landing/LandingPage';
import RedFlag from '../components/containers/incidents/RedFlag';
import Profile from '../components/containers/dashboard/UserDashboard';
import Intervention from '../components/containers/incidents/Intervention';
import NewRedflag from '../components/containers/incidents/NewRedFlag';
import NewIntervention from '../components/containers/incidents/NewIntervention';
import Admin from '../components/containers/dashboard/AdminDashboard';
import SingleRedFlag from '../components/containers/incidents/SingleRedFlag';
import SingleIntervention from '../components/containers/incidents/SingleIntervention';
import Notfound from '../components/containers/incidents/NotFound';

// require('dotenv').config();

const Routes = () => {
  return (
    <Switch>
      <Route path="/" component={LandingPage} exact />
      <Route path="/red-flags/" component={RedFlag} exact />
      <Route path="/interventions" component={Intervention} />
      <Route path="/dashboard" component={Profile} />
      <Route path="/new-red-flag" component={NewRedflag} exact />
      <Route path="/new-intervention" component={NewIntervention} />
      <Route path="/red-flags/:id?" component={SingleRedFlag} />
      <Route path="/intervention/:id?" component={SingleIntervention} />
      <Route path="/admin" component={Admin} />
      <Route component={Notfound} />
    </Switch>
  );
};

export default Routes;
