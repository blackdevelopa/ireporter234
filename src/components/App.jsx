/* eslint-disable react/jsx-filename-extension */
/* eslint-disable react/prefer-stateless-function */
import React, { Component } from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import { ToastContainer } from 'react-toastify';
import Routes from '../routes/Routes';
import classes from './style/index.css';
import store from '../store/store';
import 'react-toastify/dist/ReactToastify.css';

class App extends Component {
  render() {
    return (
      <Provider store={store}>
        <ToastContainer />
        <Router>
          <Routes className={classes.body} />
        </Router>
      </Provider>
    );
  }
}

export default App;
