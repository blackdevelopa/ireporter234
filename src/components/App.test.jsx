import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from './App';
import Routes from '../routes/Routes';

const wrapper = mount(<App />);
describe('<App/> rendering', () => {
  it('should render one <Routes> component', () => {
    expect(wrapper.find(Routes)).toHaveLength(1);
  });
  it('matches the snapshot', () => {
    const tree = shallow(<App />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
