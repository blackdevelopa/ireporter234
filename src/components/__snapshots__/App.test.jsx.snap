// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> rendering matches the snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <ToastContainer
    autoClose={5000}
    bodyClassName={null}
    className={null}
    closeButton={
      <CloseButton
        ariaLabel="close"
      />
    }
    closeOnClick={true}
    draggable={true}
    draggablePercent={80}
    hideProgressBar={false}
    newestOnTop={false}
    pauseOnFocusLoss={true}
    pauseOnHover={true}
    position="top-right"
    progressClassName={null}
    progressStyle={null}
    rtl={false}
    style={null}
    toastClassName={null}
    transition={[Function]}
  />
  <BrowserRouter>
    <Routes />
  </BrowserRouter>
</Provider>
`;
